earthaccess/store.py

```diff
@@ -466,7 +466,7 @@ def get(
             List of downloaded files
         """
         if local_path is None:
-            local_path = os.path.join(
+            local_path = Path(
```

Comment:
`local_path`, when passed to any of the `_get` dispatch methods, is expected to be a string.

Either we should change the `_get` methods to take a Path or this needs to be made a string.

Reply:
Switching the called methods to taking a Path object looks straightforward, so I'd lean that way.

Reply:
>Switching the called methods to taking a Path object looks straightforward, so I'd lean that way.

I like this too :)

Reply:
I'll update the methods to accept a Path objects - thanks for pointing that out!

Reply:
Thanks for your work making _earthaccess_ better :)